src/tsdb.c

```diff
@@ -201,23 +220,109 @@ size_t SeriesGetNumSamples(const Series *series) {
     return numSamples;
 }
 
+static void upsertHandleRules(Series *series, AddCtx *aCtx) {
```

Comment:
This looks like its replicates what `handleCompaction` does, what am I missing?

Reply:
`upsertHandleRules` handles downsampling in previous windows and skip inserts/updates in latest window.
'handleCompaction' handles ongoing downsampling in current window until it finishes.